### PR TITLE
fix tour page issue

### PR DESCRIPTION
**Pull request by reactexcel**

### CODE REVIEW

File: src/pages/data/tourDataType.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Component } from '@angular/core';
```
Comment (reactexcel) [inline src/pages/data/tourDataType.ts]: change file location to pages/tour/tourDataType.ts

File: src/pages/takeTour/tour.ts [hunk 2]
```diff
@@ -2,55 +2,55 @@ import { Component, OnInit, ViewChild} from '@angular/core';
...
-    constructor(public local: Storage, public navCtrl: NavController, public viewCtrl: ViewController, public _formService: FormService) { }
-
+    constructor(private _tourService: TourService, public local: Storage, public navCtrl: NavController, public viewCtrl: ViewController, public _formService: FormService) { }
     ngOnInit() {
-        this.local.get('store_id').then((value: any) => {
-            this.store_id = JSON.parse(value);
-        });
-        this.getTour();
-
+        let res = this._tourService.setData();
```
Comment (reactexcel) [inline src/pages/takeTour/tour.ts]: setData() is to set something. but it looks like you are using it to get some data?

File: src/providers/config/config.ts [hunk 3]
```diff
@@ -1,5 +1,12 @@
 
-export const config={
-    "APP_ID":"com.tethr",
-    "api_Url":"http://144.76.34.244:7000/"
+export const config = {
+    "APP_ID": "com.tethr",
+    "api_Url": "http://144.76.34.244:7000/",
+    "clientid":"1031859043774-igp8cfuhmbu0tfoc4fsoh7hsvhcesubc.apps.googleusercontent.com1031859043774-kqq13o4p0tu4etb7977ji042gf9sg071.apps.googleusercontent.com",
+    "mySlideOptions": {
```
Comment (reactexcel) [inline src/providers/config/config.ts]: change mySlideOptions to tourPageSliderOptions so it clear where this is used

File: src/providers/config/config.ts [hunk 4]
```diff
@@ -1,5 +1,12 @@
 
-export const config={
-    "APP_ID":"com.tethr",
-    "api_Url":"http://144.76.34.244:7000/"
+export const config = {
+    "APP_ID": "com.tethr",
+    "api_Url": "http://144.76.34.244:7000/",
+    "clientid":"1031859043774-igp8cfuhmbu0tfoc4fsoh7hsvhcesubc.apps.googleusercontent.com1031859043774-kqq13o4p0tu4etb7977ji042gf9sg071.apps.googleusercontent.com",
```
Comment (reactexcel) [inline src/providers/config/config.ts]: change client id to google_clientid

File: src/providers/tour-service/tourService.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        });
+    }
+    getData() {
+        let body = { store_id: this.store_id };
+        this._formService.api("web/config", body).subscribe((res) => {
+            this.data = JSON.parse(res.body).data;
+            return new Promise((resolve: any, reject: any) => resolve(this.data));
+        })
+    }
+    setData() {
```
Comment (reactexcel) [inline src/providers/tour-service/tourService.ts]: format data

File: src/providers/tour-service/tourService.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    data: any= { };
+    store_id: string;
+    constructor(public local: Storage, public _formService: FormService) { }
+
+    ngOnInit() {
+        this.local.get('store_id').then((value: any) => {
+            this.store_id = JSON.parse(value);
+        });
+    }
+    getData() {
```
Comment (reactexcel) [inline src/providers/tour-service/tourService.ts]: change getData() to getAppConfig()

File: src/providers/tour-service/tourService.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        let body = { store_id: this.store_id };
+        this._formService.api("web/config", body).subscribe((res) => {
+            this.data = JSON.parse(res.body).data;
+            return new Promise((resolve: any, reject: any) => resolve(this.data));
```
Comment (reactexcel) [inline src/providers/tour-service/tourService.ts]: this is not how promise is returned. if u use getData in any of your component, the return value will be null of this function not the promise

File: src/providers/tour-service/tourService.ts (hunk 5 above)
Comment (reactexcel) [inline src/providers/tour-service/tourService.ts]: what is purpose of this function?

File: src/providers/tour-service/tourService.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import { Injectable, OnInit}    from '@angular/core';
+import {FormService } from './../form-service/form-service';
+import { Storage } from '@ionic/storage';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/observable/throw';
+import 'rxjs/add/operator/toPromise';
+import keys from 'lodash/keys';
+@Injectable()
+export class TourService implements OnInit {
```
Comment (reactexcel) [inline src/providers/tour-service/tourService.ts]: the entire service is for web/config api.

change name from TourService to AppConfig

File: src/pages/takeTour/tour.ts [hunk 9]
```diff
@@ -2,55 +2,55 @@ import { Component, OnInit, ViewChild} from '@angular/core';
...
+            this._tourService.setData().then((res) => {
+                this.getTour(res);
+            })
+        }
     }
     close() {
         this.viewCtrl.dismiss();
     }
-    getTour() {
+    getTour(res) {
```
Comment (reactexcel) [inline src/pages/takeTour/tour.ts]: the name of function is getTour() but it doesnt return anything???

you need to name functions properly

File: src/pages/data/tourDataType.ts (hunk 1 above)
Comment (upexcel) [inline src/pages/data/tourDataType.ts]: done

File: src/pages/takeTour/tour.ts (hunk 2 above)
Comment (upexcel) [inline src/pages/takeTour/tour.ts]: fixed

File: src/pages/takeTour/tour.ts (hunk 9 above)
Comment (upexcel) [inline src/pages/takeTour/tour.ts]: fixed

File: src/providers/config/config.ts (hunk 4 above)
Comment (upexcel) [inline src/providers/config/config.ts]: fixed

File: src/providers/tour-service/tourService.ts (hunk 8 above)
Comment (upexcel) [inline src/providers/tour-service/tourService.ts]: fixed

File: src/providers/tour-service/tourService.ts (hunk 6 above)
Comment (upexcel) [inline src/providers/tour-service/tourService.ts]: fixed

File: src/providers/tour-service/tourService.ts (hunk 5 above)
Comment (upexcel) [inline src/providers/tour-service/tourService.ts]: fixed 
setData is rename by getData it coll in tour page and its return the value of web/config api which is set in root page

File: src/providers/tour-service/tourService.ts (hunk 7 above)
Comment (upexcel) [inline src/providers/tour-service/tourService.ts]: unable to understand .

**Pull request merged**